indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/ParallelIndexSupervisorTask.java

```diff
@@ -1817,6 +1824,11 @@ static Map<String, Object> getTaskReport(final OverlordClient overlordClient, fi
     }
   }
 
+  private InputSource getInputSource()
+  {
+    return inputSourceRef.get();
```

Comment:
The suggestion was to set this here if not already set, so that we are guaranteed to never return a null value.

Reply:
fixed

Reply:
fixed